HomeTask3/partitionOn.js

```diff
@@ -0,0 +1,33 @@
+//https://www.codewars.com/kata/partition-on
+/**
+ * 
+ * @param {number} predicate the condition on which partioning should be done on
+ * @param {array} array the array that needs to be partitioned
+ * 
+ * @returns {number} boundary- the index of the second partition's first element
+ */
+function partitionOn(predicate, array) {
+    const result = array.reduce((obj, item) => {
+        obj[predicate(item) + "Array"].push(item);
```

Comment:
Please change this. This is not readable

Reply:
done using if else